include/swift/Runtime/Win32.h

```diff
@@ -0,0 +1,55 @@
+//===--- Win32.h - Win32 utility functions ----------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Utility functions that are specific to the Windows port.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_RUNTIME_WIN32_H
+#define SWIFT_RUNTIME_WIN32_H
+
+#ifdef _WIN32
+
+#include "swift/shims/Visibility.h"
+
+#include <wchar.h>
+
+namespace swift {
+namespace win32 {
+
+/// Convert a wide string to UTF-8.
+///
+/// @param str The string to convert.
+///
+/// @returns The string, converted to UTF-8.  The caller is responsible
+///          for freeing this string with @c free() when done with it.
+///
+/// If @a str cannot be converted to UTF-8, a fatal error occurs.
+SWIFT_RUNTIME_STDLIB_INTERNAL
+char *copyUTF8FromWide(const wchar_t *str);
+
+/// Convert a UTF-8 string to a wide string.
+///
+/// @param str The string to convert.
+///
+/// @returns The string, converted to UTF-16.  The caller is responsible
+///          for freeing this string with @c free() when done with it.
+///
+/// If @a str cannot be converted to UTF-16, a fatal error occurs.
+SWIFT_RUNTIME_STDLIB_INTERNAL
+wchar_t *copyWideFromUTF8(const char *str);
```

Comment:
Can we call this `copyUTF16fromUTF8`?  It really is UTF16 encoded (even if you do something like wide is 32-bit).

Reply:
I slightly prefer "Wide" in this context because naming it "UTF16" means people might think this was some kind of generic UTF-8 to UTF-16 interface, which it very much isn't. It's specifically here to convert Swift strings for the Windows wide character APIs.

I know that the Windows wide character encoding is in practice UTF-16 (mostly… sometimes it's UCS-2), but the goal here isn't to provide a general purpose character encoder.